### PR TITLE
make the AERONET reading more fault tolerable; add possibility to read gzipped files

### PR DESCRIPTION
**Pull request by jgriesfeller**
Basically a new try for a fix of #518 done because I messed up the branch after @avaldebe merged the source run through black and isort.
No different functionality compared to #527

### CODE REVIEW

File: tests/io/test_read_aeronet_sunv3_all_points.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+def test_read_add_common_meta(reader):
+    files = reader.files
+    data = reader.read("od550aer", files=files, common_meta={"bla": 42})
+    assert all("bla" in x for x in data.metadata.values())
+
+
+if __name__ == "__main__":
+    import sys
+
+    pytest.main(sys.argv)
```
Comment (avaldebe) [inline tests/io/test_read_aeronet_sunv3_all_points.py:52]: do not add unnecessary entry points, use the IDE to run specific tests

File: tests/io/test_read_aeronet_sunv3_all_points.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+"""
+Created on 20211214
+"""
```
Comment (avaldebe) [inline tests/io/test_read_aeronet_sunv3_all_points.py:4]: not needed, git knows when the file was created and who did it

File: tests/io/test_read_aeronet_sunv3_all_points.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import os
+
+import numpy as np
+import numpy.testing as npt
```
Comment (avaldebe) [inline tests/io/test_read_aeronet_sunv3_all_points.py:8]: unused imports

File: pyaerocom/exceptions.py [hunk 4]
```diff
@@ -22,6 +22,11 @@
 """
 
 
+class AeronetReadError(IOError):
+    # Aeronet reading failed somehow
+    pass
```
Comment (avaldebe) [inline pyaerocom/exceptions.py:27]: can you write a test that raises this exception?

**Review by avaldebe — COMMENTED** [review]
I only have small suggestions, mostly a test that raises the new exception.
You can go ahead and merge if you think I'm nitpicking.

File: tests/io/test_read_aeronet_sunv3_all_points.py (hunk 1 above)
Comment (jgriesfeller) [reply tests/io/test_read_aeronet_sunv3_all_points.py:52]: I always thought pytest needed that. Apparently it does not. Removed.

Comment (avaldebe) [reply tests/io/test_read_aeronet_sunv3_all_points.py:52]: `pytest` only need the file to be called `test_*.py` and the test functions to be called `test_something`.

This entry point it is possible to run all the tests on the file with

``` bash
python path/to/test_file.py
```

the same can be achieved without the entry point with

``` bash
pytest path/to/test_file.py
```

However, I think that those entry points were added to run the tests from an IDE which does not support running the tests directly. This way running the file from the IDE will run the tests.

**Review by avaldebe — APPROVED** [review]
:+1: ready to merge

**Pull request merged**